app/scripts/views/settings.js

```diff
@@ -120,9 +120,23 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       this.logScreen();
     },
 
+    render: function () {
+      // The `settings` class is added here instead of in `beforeRender`
+      // to eliminate jerkiness. beforeRender is only called after an XHR
+      // request is made to determine the user's authentication status. On slow
+      // connections this allows the Firefox logo and the white screen used
+      // elsewhere to be displayed while the user's status is being determined.
+      // Adding the `settings` class before the XHR request ensures the Firefox
+      // logo is not displayed. If the user is not authenticated,
+      // `beforeDestroy` will take care of removing the class.
+      // See #3053
+      $('body').addClass('settings');
+
+      return BaseView.prototype.render.apply(this, arguments);
```

Comment:
I'm wondering if we should make this a generic feature, like the view could declare a `layout` property (or similar) that `BaseView` adds as a className to `body` early on. Thoughts?